### PR TITLE
Log OSIE failures to tinkerbell using a bash trap function

### PR DESCRIPTION
**Pull request by ScottGarman**
Similar to what we do in osie-runner, this sets up a bash trap
function named autofail() which logs to tinkerbell an error message
based on the `autofail_reason` variable, which we update in various
sections of the script to reflect which code section the failure
occurred.

### CODE REVIEW

**Review by mikemrm — DISMISSED** [review]
LGTM!

File: docker/scripts/osie.sh [hunk 1]
```diff
@@ -55,6 +55,17 @@ if [[ $state == 'osie.internal.check-env' ]]; then
...
+# On errors, run autofail() before exiting
+autofail_reason='unknown, just started osie'
+function autofail() {
+	if [[ "${autofail_reason}" == "SUCCESS" ]]; then
+		return 0
+	fi
+	puttink "${tinkerbell}" phone-home '{"type":"failure", "reason":"'"${autofail_reason}"'"}'
+	print_stack_trace
+}
+trap autofail EXIT
```
Comment (mmlb) [inline docker/scripts/osie.sh:68]: change this to `ERR` and you can do away with the `== "SUCCESS"` handling

File: docker/scripts/osie.sh [hunk 2]
```diff
@@ -587,3 +609,4 @@ EOF
 fi
 
 chmod +x /statedir/cleanup.sh
+autofail_reason='SUCCESS'
```
Comment (mmlb) [inline docker/scripts/osie.sh]: can go away with `trap ... ERR`

**Review by mmlb — CHANGES_REQUESTED** [review]

**Comment by ScottGarman** [comment]
Changing this to an ERR trap isn't working, I'll research this more tomorrow.

**Comment by ScottGarman** [comment]
Manny and I discussed the trap on EXIT vs. ERR issue this morning, and I've implemented the solution we came to. This is ready for final review and merge now.

**Review by mmlb — APPROVED** [review]

**Pull request merged**